pkgs/development/web/nodejs/v22.nix

```diff
@@ -10,8 +10,24 @@ buildNodejs {
   inherit enableNpm;
   version = "22.14.0";
   sha256 = "c609946bf793b55c7954c26582760808d54c16185d79cb2fb88065e52de21914";
-  patches = [
+  patches = (if (stdenv.hostPlatform.emulatorAvailable buildPackages) then [
     ./configure-emulator.patch
+  ] else [
+    (fetchpatch2 {
+      url = "https://raw.githubusercontent.com/buildroot/buildroot/2f0c31bffdb59fb224387e35134a6d5e09a81d57/package/nodejs/nodejs-src/0003-include-obj-name-in-shared-intermediate.patch";
+      hash = "sha256-3g4aS+NmmUYNOYRNc6UMJKYoaTlpP5Knt9UHegx+o0Y=";
+    })
+  ]) ++ lib.optionals (stdenv.hostPlatform != stdenv.buildPlatform && stdenv.hostPlatform.isFreeBSD) [
+    # This patch is concerning.
+    # https://github.com/nodejs/node/issues/54576
+    # It is only supposed to affect clang >= 17, but I'm seeing it on clang 19.
+    # I'm keeping the predicate for this patch pretty strict out of caution,
+    # so if you see the error it's supposed to prevent, feel free to loosen it.
+    (fetchpatch2 {
+      url = "https://raw.githubusercontent.com/rubyjs/libv8-node/62476a398d4c9c1a670240a3b070d69544be3761/patch/v8-no-assert-trivially-copyable.patch";
+      hash = "sha256-hSTLljmVzYmc3WAVeRq9EPYluXGXFeWVXkykufGQPVw=";
```

Comment:
For non-upstreamable patches, I think it makes more sense to vendor them in the repo rather than fetching them

Reply:
I think it makes more sense for us to use another distro's patches if we're not super clear on the patch's long term situation because we can refer back to the other distro in order to see how they adjust the patch moving forward. If you think this is a silly/counterproductive/antisocial/etc argument I will gladly vendor it though!